File: UseGL/app/src/main/java/org/example/usegl/MainActivity.java
package org.example.usegl;

import android.support.v7.app.ActionBarActivity;
import android.os.Bundle;
import android.util.Log;

public class MainActivity extends ActionBarActivity {

	@Override protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		_view = new GLSurfaceViewImpl(getApplication());
		setContentView(_view);
		Log.d(TAG, "onCreate()");
	}

	@Override protected void onResume() {
		Log.d(TAG, "onResume()");
		super.onResume();
		_view.onResume();
	}

	@Override protected void onPause() {
		Log.d(TAG, "onPause()");
		super.onPause();
		_view.onPause();
	}

	private GLSurfaceViewImpl _view;
	private static String TAG = "MainActivity";
}

File: kotlin/DatePickerDialog/app/src/main/java/com/example/datepickerdialog/MainActivity.kt
package com.example.datepickerdialog

import android.app.DatePickerDialog
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import java.util.*

class MainActivity : AppCompatActivity() {

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.activity_main)

		val today = Calendar.getInstance()
		val d = DatePickerDialog(this,
			DatePickerDialog.OnDateSetListener { view, year, month, dayOfMonth ->
				Toast.makeText(this, "$dayOfMonth/$month/$year", Toast.LENGTH_SHORT).show()
			}, today.get(Calendar.YEAR), today.get(Calendar.MONTH), today.get(Calendar.DAY_OF_MONTH))
		d.show()
	}
}
